elliott/elliottlib/cli/release_snapshot_cli.py

```diff
@@ -0,0 +1,195 @@
+        self.runtime.initialize()
+        if self.runtime.konflux_db is None:
+            raise RuntimeError('Must run Elliott with Konflux DB initialized')
+        if self.for_bundle:
+            self.runtime.konflux_db.bind(KonfluxBundleBuildRecord)
+        else:
+            self.runtime.konflux_db.bind(KonfluxBuildRecord)
+
+        # Ensure the Snapshot CRD is accessible
+        try:
+            await self._konflux_client._get_api(API_VERSION, KIND_SNAPSHOT)
+        except exceptions.ResourceNotFoundError:
+            raise RuntimeError(f"Cannot access {API_VERSION} {KIND_SNAPSHOT} in the cluster. Passed the right kubeconfig?")
+
+        build_records: list[KonfluxRecord] = await self.fetch_build_records()
+
+        # make sure pullspec is live for each build
+        image_info_tasks = []
+        for record in build_records:
+            image_info_tasks.append(asyncio.create_task(
+                oc_image_info_for_arch_async(
+                    record.image_pullspec,
+                    registry_username=os.environ.get('KONFLUX_ART_IMAGES_USERNAME'),
+                    registry_password=os.environ.get('KONFLUX_ART_IMAGES_PASSWORD'),
+                )))
+        image_infos = await asyncio.gather(*image_info_tasks, return_exceptions=True)
+        errors = [(record, result) for record, result in zip(build_records, image_infos)
+                  if isinstance(result, BaseException)]
+        if errors:
+            for record, ex in errors:
+                record: KonfluxRecord
+                LOGGER.error("Failed to inspect nvr %s pullspec %s: %s", record.nvr, record.image_pullspec, ex)
+            raise RuntimeError("Failed to inspect build pullspecs")
+
+        snapshot_obj = await self.new_snapshot(build_records)
+        created = await self._konflux_client._create(snapshot_obj)
+        print(created)
+
+    async def new_snapshot(self, build_records) -> dict:
+        major, minor = self.runtime.get_major_minor()
+        timestamp = datetime.strftime(datetime.now(tz=timezone.utc), "%Y%m%d%H%M")
+        snapshot_name = f"ose-{major}-{minor}-{timestamp}"
+        application_name = KonfluxImageBuilder.get_application_name(self.runtime.group)
+
+        # make sure application exists
+        await self._konflux_client._get(API_VERSION, KIND_APPLICATION, application_name)
+
+        async def _comp(record):
+            comp_name = KonfluxImageBuilder.get_component_name(application_name, record.name)
+
+            # make sure component exists
+            await self._konflux_client._get(API_VERSION, KIND_COMPONENT, comp_name)
+
+            source_url = record.source_repo
+            revision = record.commitish
+
+            # pullspec is in the form of registry/image:tag
+            # we need to provide snapshot with digest which is stored in image_tag
+            image = record.image_pullspec.split(':')
+            digest = record.image_tag
+            pullspec = f"{image[0]}@sha256:{digest}"
```

Comment:
I think we should pass in the manifest list SHA and not the digest?

Reply:
digest == manifest list SHA :)

Reply:
```
(asdas) asdas-mac:asdas$ oc image info quay.io/redhat-user-workloads/ocp-art-tenant/art-images@sha256:0168129b467c71770845fd95bfcf4435efa5280a12fe1fd6c368c29c5b3008b4 --filter-by-os amd64
Name:          quay.io/redhat-user-workloads/ocp-art-tenant/art-images@sha256:0168129b467c71770845fd95bfcf4435efa5280a12fe1fd6c368c29c5b3008b4
Digest:        sha256:75b5a5df251a53992bf05593c6daee6a65457bc1869969fef6fd3474735a36f5
Manifest List: sha256:0168129b467c71770845fd95bfcf4435efa5280a12fe1fd6c368c29c5b3008b4
```

Reply:
In the DB, we only store the manifest list digest and do not store per arch digest. That is what is being fetched here and constructed in the pullspec.

Reply:
Maybe there's some terminology confusion here. hash/digest/sha are interchangeably being used here .. hash/digest are more generic, sha more specific since it's the hashing algo. If we or something else used md5 hashing algorithm, we would call it md5 digest or md5 hash. here it is sha digest or sha hash. 

In regards to the DB, the field names are confusing. I would expect the `image_pullspec` to be the one like `quay.io .. @sha256:foobar` and `image_tag` to have nvr_like tag.. but it's the other way around right now :/

Reply:
yeah I was talking about the naming. I guess its fine